nodebuilder/state/state.go

```diff
@@ -79,47 +81,154 @@ type Module interface {
 // API is a wrapper around Module for the RPC.
 // TODO(@distractedm1nd): These structs need to be autogenerated.
 type API struct {
```

Comment:
I would actually keep them in the same file as long as Module is at the top but long term they will end up in their own files once I create autogeneration for these structs - should I bite the bullet and do it now, or can it stay until then? Not very opinionated here

Reply:
@renaynay and I decided in sync to delay this until autogen is done, primarily because doing so now would make rebasing the stacked PRs a nightmare for little direct benefit

Reply:
👌🏻